src/Umbraco.Core/Services/Implement/MediaService.cs

```diff
@@ -1139,6 +1147,26 @@ public bool Sort(IEnumerable<IMedia> items, int userId = Constants.Security.Supe
             }
 
             return true;
+
+        }
+
+        public ContentDataIntegrityReport CheckDataIntegrity(ContentDataIntegrityReportOptions options)
+        {
+            using (var scope = ScopeProvider.CreateScope(autoComplete: true))
+            {
+                scope.WriteLock(Constants.Locks.MediaTree);
+
+                var report = _mediaRepository.CheckDataIntegrity(options);
+
+                if (report.FixedIssues.Count > 0)
+                {
+                    //The event args needs a content item so we'll make a fake one with enough properties to not cause a null ref
+                    var root = new Models.Media("root", -1, new MediaType(-1)) { Id = -1, Key = Guid.Empty };
```

Comment:
We don't want to provide the actual changed items, we want to change ALL of the items from the root. TreeChange event args are not supposed to include individual items because that would be enormously expensive (we've tried that in the past), these events tell the cache refreshers how much to change on what branch given a single node.